sdk/src/main/java/com/endare/adhese/sdk/AdheseOptions.java

```diff
@@ -101,6 +117,24 @@ public Builder withCookieMode(CookieMode cookieMode) {
             return this;
         }
 
+        public Builder addCustomParameter(String key, String... values) {
+            return this.addCustomParameter(key, Arrays.asList(values));
+        }
+
+        public Builder addCustomParameter(String key, Collection<String> values) {
+            if (key == null || values == null || key.length() != 2 || values.isEmpty()) {
+                throw new IllegalArgumentException("To add a valid custom parameter, your key must be two chars long and have at least one value.");
+            }
+
+            if (this.options.customParameters.containsKey(key)) {
+                this.options.customParameters.get(key).addAll(values);
+            } else {
+                this.options.customParameters.put(key, new HashSet<>(values));
+            }
+
+            return this;
```

Comment:
It's the responsibility of the caller to html escape the values ?

Reply:
Perhaps not. I'll add that, and the 'illegal' chars validation

Reply:
but what about the other target values? were they escaped?

Reply:
no they are not. So maybe it is their responsibility